packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -462,3 +473,31 @@ class StackDependentToken extends cdk.Token {
     return this.fn(context.scope.node.stack);
   }
 }
+
+class ServicePrincipalToken extends cdk.Token {
+  constructor(private readonly service: string,
+              private readonly opts: ServicePrincipalOpts) {
+    super();
+  }
+
+  public resolve(ctx: cdk.ResolveContext) {
+    const region = this.opts.region || ctx.scope.node.stack.region;
+    const fact = RegionInfo.get(region).servicePrincipal(this.service);
+    if (!fact) {
+      ctx.scope.node.addWarning(`No regional service principal found for ${this.service} in ${region}.`);
```

Comment:
What if I want to use internal service principals? Am I going to get this warning every time, even though everything is okay? Can we maybe only generate this warning if it looks like I'm trying to look one up? (For example, if the service principal contains the string `"amazonaws"`  or doesn't contain a `"."` ?

Reply:
At this stage, I would generate the warning unconditionally, and we can look into reducing "warning verbosity" once we see it being an issue?